game/game_map.py

```diff
@@ -1,24 +1,25 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Iterable, Iterator, Optional
+from typing import TYPE_CHECKING, Iterable, Iterator, Optional, Set
 
-from tcod.console import Console
 import numpy as np
+import tcod
 
-from entity import Actor, Item
-import tile_types
+import game.tiles
 
 if TYPE_CHECKING:
-    from engine import Engine
-    from entity import Entity
+    import game.engine
+    import game.entity
 
 
 class GameMap:
-    def __init__(self, engine: Engine, width: int, height: int, entities: Iterable[Entity] = ()):
+    def __init__(
+        self, engine: game.engine.Engine, width: int, height: int, entities: Iterable[game.entity.Entity] = ()
+    ):
         self.engine = engine
         self.width, self.height = width, height
-        self.entities = set(entities)
-        self.tiles = np.full((width, height), fill_value=tile_types.wall, order="F")
+        self.entities: Set[game.entity.Entity] = set(entities)
```

Comment:
Possible redundant type-hint here. `set(entities)` already defines this hint.

Even if you wanted to explicitly hint this, `typing.Set` is known to be deprecated, replaced by `collections.abc.Set`.